pkg/solana/chainreader/event_read_binding.go

```diff
@@ -115,6 +158,56 @@ func (b *eventReadBinding) QueryKey(
 	return sequences, nil
 }
 
+func (b *eventReadBinding) normalizeParams(value any, itemType string) (any, error) {
+	offChain, err := b.codec.CreateType(itemType, true)
+	if err != nil {
+		return nil, fmt.Errorf("%w: failed to create type: %w", types.ErrInvalidType, err)
+	}
+
+	// params can be a singular primitive value, a map of values, or a struct
+	// in the case that the input params are presented as a map of values, apply the values to the off-chain type
+	// with solana hooks
+	if err = codec.MapstructureDecode(value, offChain); err != nil {
+		return nil, fmt.Errorf("%w: failed to decode offChain value: %s", types.ErrInternal, err.Error())
+	}
+
+	return offChain, nil
+}
+
+func (b *eventReadBinding) extractFilterSubkeys(offChainValue any) (query.Expression, error) {
```

Comment:
If I understand this correctly,`offChainValue` here is a map of key-value params right?  If so, would be easier to read if we renamed it to something like `offChainParams`.

Reply:
I can change it if it adds some clarity.